utils/streams.js

```diff
@@ -0,0 +1,127 @@
+import commander from 'commander';
+import through from 'through2';
+import fs from 'fs';
+import path from 'path';
+import { promisify } from 'util';
+import csvjson from 'csvjson';
+const readFileAsync = promisify(fs.readFile);
+const readDirAsync = promisify(fs.readdir);
+
+function streams() {
+    commander.option('-a, --action <string>', 'action name taken')
+        .option('-f, --file <string>', 'file will read')
+        .option('-p, --path <string>', 'path for css folder')
+        .parse(process.argv);
+    executeTheAction();
+}
+const actionList = {reverse, transform, outputFile, convertFromFile, convertToFile};
+function executeTheAction() {
+    if (process.argv.length < 3) {
+        commander.help();
+    } else if (process.argv[2] == '-h' || process.argv[2] == '--help') {
+        commander.help();       
+    } else {
+        if (!!commander.action) {
+            if(actionList[commander.action] != undefined) {
+                actionList[commander.action]();
+            } else{
+                console.log('Err!! Invalid arguement passed. Please choose correct action!!!');
+            }
+        } else {
+            console.log('Err!! No argument passed. Please choose correct action!!!');
+            process.exit();
+        }
+    }
+}
+
+function reverse() {
+    process.stdin.on('data', function (data) {
+        const splitString = data.toString().split('').reverse().join('');
+        process.stdout.write(splitString);
+    });
+}
+
+//Transform stream to uppercase
+function transform() {
+    const transToUpperCase = (data, encoding, next) => {
```

Comment:
When you changed function declaration into arrow function the lexical context is not the same. `this.push` cracks

Reply:
Apologies, I didn't notice it.